frontend/src/components/applyForm/widgets/MultiSelectWidget.tsx

```diff
@@ -0,0 +1,216 @@
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+} from "@rjsf/utils";
+
+import React, { useCallback, useMemo, useRef } from "react";
+import {
+  ComboBox,
+  ErrorMessage,
+  FormGroup,
+  type ComboBoxOption,
+  type ComboBoxRef,
+} from "@trussworks/react-uswds";
+
+import { UswdsWidgetProps } from "src/components/applyForm/types";
+import { Pill } from "src/components/Pill";
+import { DynamicFieldLabel } from "./DynamicFieldLabel";
+import { getLabelTypeFromOptions } from "./getLabelTypeFromOptions";
+
+function toArray(selectedValue: unknown): string[] {
+  if (Array.isArray(selectedValue)) return selectedValue.map(String);
+  if (selectedValue == null || selectedValue === "") return [];
+  return [String(selectedValue)];
+}
+
+export default function MultiSelect<
+  T = unknown,
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = never,
+>({
+  id,
+  disabled,
+  readonly,
+  required,
+  schema,
+  value,
+  options = {},
+  autofocus = false,
+  rawErrors = [],
+  onChange = () => ({}),
+  onBlur = () => ({}),
+  onFocus = () => ({}),
+}: UswdsWidgetProps<T, S, F>) {
+  const { title, description, maxItems, minItems } = schema;
+
+  const { enumOptions: opts, enumDisabled } = (options ?? {}) as {
+    enumOptions?: ComboBoxOption[];
+    enumDisabled?: Array<string | number>;
+  };
+  // Future:
+  // allows us to inforce the minimum number
+  // we can later pass it as prop if design likes the idea
+  // this will be used for some dropdowns that have the possibility of having one option
+  // example would be starting an application and it is defaulted to "individual" and no other options are available
+  const enforceMinInUI = false;
+  const labelType = getLabelTypeFromOptions(options?.["widget-label"]);
+  const allOptions: ComboBoxOption[] = useMemo(() => opts ?? [], [opts]);
+
+  const [selected, setSelected] = React.useState<string[]>(toArray(value));
+
+  const comboRef = useRef<ComboBoxRef>(null);
+
+  const maxSelections = typeof maxItems === "number" ? maxItems : 3;
+  const minSelectionsEff = typeof minItems === "number" ? minItems : 0;
+  const atCap = selected.length >= maxSelections;
+
+  const isOptDisabled = useCallback(
+    (opt: ComboBoxOption) =>
+      !!enumDisabled &&
+      (enumDisabled.includes(opt.label) || enumDisabled.includes(opt.value)),
+    [enumDisabled],
+  );
+
+  const availableOptions = useMemo(
+    () =>
+      allOptions.filter(
+        (o) => !selected.includes(String(o.value)) && !isOptDisabled(o),
+      ),
+    [allOptions, selected, isOptDisabled],
+  );
+
+  const error = rawErrors.length ? true : undefined;
+  const describedBy = error
+    ? `error-for-${id}`
+    : title
+      ? `label-for-${id}`
+      : undefined;
+
+  const syncUpstream = (next: string[]) => {
+    setSelected(next);
+    onChange(
+      enumOptionsValueForIndex<S>(next, allOptions, undefined) as unknown as T,
+    );
+  };
+
+  const addByLabelOrValue = (raw: string): boolean => {
```

Comment:
it doesn't look like we're using the return value here - I think it's potentially confusing to return a boolean in that case. Can we remove that?